react-1/src/App.js

```diff
@@ -2,20 +2,49 @@
 
 import React, { Component } from 'react';
 
-import PresentationList from './PresentationList';
+import PresentationList, { type Agenda } from './PresentationList';
 
 import logo from './logo.svg';
 import './App.css';
 
-class App extends Component<{||}> {
+const FETCH_INTERVAL_MS = 5000;
+
+type State = {|
+  agenda: Agenda | null,
+|};
+
+class App extends Component<{||}, State> {
+  _timeoutId: TimeoutID | null = null;
+  state = { agenda: null };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillUnmount() {
+    if (this._timeoutId !== null) {
+      clearTimeout(this._timeoutId);
+      this._timeoutId = null;
+    }
+  }
+
+  async fetchData() {
+    const fetchResult = await fetch('/agenda.json');
+    const result = await fetchResult.json();
+    this.setState({ agenda: result });
+    this._timeoutId = setTimeout(() => this.fetchData(), FETCH_INTERVAL_MS);
```

Comment:
You must wrap the fetching in a `try/catch` otherwise a failed fetch attempt will stop the loop?

Reply:
done :)